### PR TITLE
DEV/MNT: migrate to GitHub Actions and pyproject.toml [LCLSPC-603]

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Upgrading by way of https://github.com/pcdshub/pcds-python-migration-tools
* Conda jobs will not pass due to missing epicsmacrolib on conda-forge. Tests were previously passing due to custom travis 
config
* Python 3.11 testing on pip now passes with new tag of epicsmacrolib (regenerated Cython bindings)
* We'll move forward with pip-only for now
* Required-to-pass jobs have been adjusted based on the above
* Documentation building fixed after some mkdocstrings changes and a `Makefile` was added to conform to our standard CI workflow config

## Motivation and Context
* Migrating because we want to migrate our projects from Travis CI for one
* Additionally because of confusing conda build output in #151 (which this PR just drops because we can't build for conda anyway)

## How Has This Been Tested?
* GitHub Actions

## Where Has This Been Documented?
This PR and Jira [LCLSPC-603](https://jira.slac.stanford.edu/browse/LCLSPC-603)

### CODE REVIEW

**Comment by klauer** [comment]
* Python 3.11 testing on pip now passes with new tag of epicsmacrolib (regenerated Cython bindings)
* Conda jobs will not pass due to missing epicsmacrolib on conda-forge. Tests were previously passing due to custom travis config
* Documentation building fixed after some mkdocstrings changes and a `Makefile` was added to conform to our standard CI workflow config

File: pyproject.toml [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
-minversion = "6.0"
-testpaths = [
-    "whatrecord/tests",
-]
-asyncio_mode = "auto"
+build-backend = "setuptools.build_meta"
+requires = [ "setuptools>=45", "setuptools_scm[toml]>=6.2",]
+
+[project]
+classifiers = [ "Development Status :: 2 - Pre-Alpha", "Natural Language :: English", "Programming Language :: Python :: 3",]
```
Comment (ZLLentz) [inline pyproject.toml:6]: Maybe a higher dev status here
```suggestion
classifiers = [ "Development Status :: 5 - Production/Stable", "Natural Language :: English", "Programming Language :: Python :: 3",]
```

**Review by ZLLentz — APPROVED** [review]
One small suggestion but LGTM

**Review by tangkong — APPROVED** [review]
LGTM

**Pull request merged**